cmd/antrea-agent/options.go

```diff
@@ -397,22 +398,32 @@ func (o *Options) setK8sNodeDefaultOptions() {
 	if o.config.HostProcPathPrefix == "" {
 		o.config.HostProcPathPrefix = defaultHostProcPathPrefix
 	}
-	if features.DefaultFeatureGate.Enabled(features.AntreaProxy) {
+	if !features.DefaultFeatureGate.Enabled(features.AntreaProxy) {
+		o.enableAntreProxy = false
+		klog.InfoS("Feature gate `AntreaProxy` is deprecated, please use option `antreaProxy.enable` to disable AntreaProxy")
+	} else {
+		o.enableAntreProxy = true
+	}
```

Comment:
I think things are getting more complicated with these code.
We should:
1. Let defaulting func only do defaulting
```
if o.config.AntreaProxy.Enable == nil {
	o.config.AntreaProxy.Enable = pointer.Bool(true)
}
if o.config.AntreaProxy.ProxyLoadBalancerIPs == nil {
	o.config.AntreaProxy.ProxyLoadBalancerIPs = pointer.Bool(true)
}
if o.config.AntreaProxy.DefaultLoadBalancerMode == "" {
	o.config.AntreaProxy.DefaultLoadBalancerMode = config.LoadBalancerModeNAT.String()
}
```
2. Let validating func do validating, calculate `o.enableAntreProxy`
```
if !features.DefaultFeatureGate.Enabled(features.AntreaProxy) {
	klog.InfoS("Feature gate `AntreaProxy` is deprecated, please use option `antreaProxy.enable` to disable AntreaProxy")
}
o.enableAntreaProxy = *o.config.AntreaProxy.Enable && !features.DefaultFeatureGate.Enabled(features.AntreaProxy)
if o.enableAntreaProxy {
...
} else {
...
}
```
3. Let run func consume only `o.enableAntreProxy`, do not calculate whether it's enabled again.